tests/autorestart_test/test_container_autorestart.py

```diff
@@ -0,0 +1,354 @@
+
+    return container_autorestart_states
+
+
+def get_disabled_container_list(duthost):
+    """
+    @summary: Get the container/service names which are disabled by default
+    @return: A list includes the names of disabled containers/services
+    """
+    disabled_containers = []
+
+    show_cmd_output = duthost.shell(CMD_FEATURE)
+    for line in show_cmd_output["stdout_lines"]:
+        container_name = line.split()[0].strip()
+        container_state = line.split()[1].strip()
+        if container_state == "disabled":
+            disabled_containers.append(container_name)
+
+    return disabled_containers
+
+
+def is_container_running(duthost, container_name):
+    """
+    @summary: Decide whether the container is running or not
+    @return:  Boolean value. True represents the container is running
+    """
+    is_running = duthost.shell("docker inspect -f \{{\{{.State.Running\}}\}} {}".format(container_name))
+    if is_running["stdout_lines"][0].strip('\n') == "true":
+        return True
+    return False
+
+
+def check_container_state(duthost, container_name, should_be_running):
+    """
+    @summary: Determine whether a container should be in running state or not
+    """
+    is_running = is_container_running(duthost, container_name)
+    if is_running == should_be_running:
+        return True
+    else:
+        return False
+
+
+def get_program_status(duthost, container_name, program_name):
+    """
+    @summary: Return the status of a program in the specified container
+    @return: "RUNNING" or "EXITING" represents the program is in the running
+             or exited status
+    """
+    program_status = None
+
+    process_list = duthost.shell("docker exec {} supervisorctl status".format(container_name))
+    for process_info in process_list["stdout_lines"]:
+        if process_info.find(program_name) != -1:
+            program_status = process_info.split()[1].strip()
+            break
+
+    return program_status
+
+
+def kill_process_by_pid(duthost, container_name, program_name, program_pid):
+    """
+    @summary: Kill a process in the specified container by its pid
+    """
+    duthost.shell("docker exec {} kill -SIGKILL {}".format(container_name, program_pid))
+
+    # Wait 30 seconds such that the container is stopped
+    wait_until(CONTAINER_STOP_THRESHOLD_SECS,
+               CONTAINER_CHECK_INTERVAL_SECS,
+               check_container_state, duthost, container_name, False)
+
+    is_running = is_container_running(duthost, container_name)
+    if is_running:
+        program_status = get_program_status(duthost, container_name, program_name)
+        # If program is still in the running state, that means it is not terminated by "kill" command
+        # and the testing will exit
+        if program_status == "RUNNING":
+            pytest.fail("Failed to stop {} process before test".format(program_name))
+
+    logger.info("{} in {} container is stopped successfully".format(program_name, container_name))
```

Comment:
Enhance message: `"Program '{}' in container '{}' was stopped successfully"`

Reply:
Updated